app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb

```diff
@@ -142,4 +142,18 @@ def available_vlans_and_hosts(options = {})
     end
     return vlans, hosts
   end
+
+  def allowed_storage_profiles(_options = {})
+    return [] if (src = resources_for_ui).blank? || src[:vm].nil?
+    @filters[:StorageProfile] ||= begin
+      template = load_ar_obj(src[:vm])
+      @values[:placement_storage_profile] = [template.storage_profile.try(:id), template.storage_profile.try(:name)]
```

Comment:
The StorageProfile is almost working like a secondary filter for datastores that user can see/use.  I.e. Even if we don't have it managed by RBAC and allow user to see all of them, it will not allow user to see/use datastores that are already limited by their RBAC role.

From that perspective, I think we are fine. Do I miss anything?